controller/program/tests/tests/add_lst.rs

```diff
@@ -0,0 +1,560 @@
+        .with_push(LstState {
+            is_input_disabled: 0,
+            pool_reserves_bump,
+            protocol_fee_accumulator_bump,
+            padding: [0u8; 5],
+            sol_value: 0,
+            mint: *mint,
+            sol_value_calculator: *expected_sol_value_calculator,
+        })
+        .build();
+
+    assert_diffs_lst_state_list(&diffs, &lst_state_list_bef, &lst_state_list_aft);
+}
+
+#[test]
+fn add_lst_jitosol_fixture() {
+    let pool_pk = Pubkey::new_from_array(POOL_STATE_ID);
+    let pool_acc = ALL_FIXTURES
+        .get(&pool_pk)
+        .expect("missing pool state fixture");
+    let pool = PoolStatePacked::of_acc_data(&pool_acc.data)
+        .unwrap()
+        .into_pool_state();
+
+    let admin = pool.admin;
+    let token_program = &TOKENKEG_ID;
+    let sol_value_calculator = &SPL_SVC;
+
+    let keys = add_lst_ix_keys_owned(
+        &admin,
+        &admin,
+        &JITOSOL_MINT.to_bytes(),
+        token_program,
+        sol_value_calculator,
+    );
+
+    let ix = add_lst_ix(&keys);
+    let mut accounts = add_lst_fixtures_accounts_opt(&keys);
+
+    upsert_account(
+        &mut accounts,
+        (
+            Pubkey::new_from_array(admin),
+            Account {
+                lamports: u64::MAX,
+                ..Default::default()
+            },
+        ),
+    );
+
+    upsert_account(
+        &mut accounts,
+        (
+            Pubkey::new_from_array(PROTOCOL_FEE_ID),
+            Account {
+                ..Default::default()
+            },
+        ),
+    );
+
+    let InstructionResult {
+        program_result,
+        resulting_accounts,
+        ..
+    } = SVM.with(|svm| svm.process_instruction(&ix, &accounts));
+
+    assert_eq!(program_result, ProgramResult::Success);
+
+    assert_correct_add(
+        &accounts,
+        &resulting_accounts,
+        &JITOSOL_MINT.to_bytes(),
+        token_program,
+        sol_value_calculator,
+    );
+}
+
+enum TestErrorType {
+    Unauthorized,
+    PoolRebalancing,
+    PoolDisabled,
+    DuplicateLst,
+    NonExecSvc,
+}
+
+const MAX_LST_STATES: usize = 10;
+
+#[allow(clippy::too_many_arguments)]
+fn add_lst_proptest(
+    pool: PoolState,
+    lsl: LstStateListData,
+    admin: [u8; 32],
+    payer: [u8; 32],
+    mint: [u8; 32],
+    token_program: [u8; 32],
+    sol_value_calculator: [u8; 32],
+    additional_accounts: impl IntoIterator<Item = PkAccountTup>,
+    error_type: Option<TestErrorType>,
+) -> TestCaseResult {
+    silence_mollusk_logs();
+
+    let LstStateListData { lst_state_list, .. } = lsl;
+
+    let keys = add_lst_ix_keys_owned(&admin, &payer, &mint, &token_program, &sol_value_calculator);
+
+    let ix = add_lst_ix(&keys);
+    let mut accounts = add_lst_fixtures_accounts_opt(&keys);
+
+    // Common upserts
+    upsert_account(
+        &mut accounts,
+        (
+            LST_STATE_LIST_ID.into(),
+            lst_state_list_account(lst_state_list),
```

Comment:
i realized from reading the [old code that the owner of LstStateList will be system program if it was empty](https://github.com/igneous-labs/S/blob/9fc17034f7b5376f27a6925baa1e65af05843cba/programs/s-controller/src/list_account.rs#L22-L51). Could you modify `lst_state_list_account` to special-case owner and lamports if length == 0?